src/app/components/daily-review/daily-review.component.ts

```diff
@@ -1,50 +1,328 @@
 
-    })
   }
+  checkIfHabitLogsExist() {
+
+    this.accountService.getCurrentDateHabitLogEntries(
+      this.currentDate,
+      this.currentUserId).
+      pipe(take(1)).
+      subscribe(logs => {
+        //save logs for later http call
+        this.currentDateHabitLogs = logs;
+
+        if (logs.length !== 0) {
+          //if logs exist for currentDate, loop through the FormControls and add the existing log values to the form
+          for (let i = 0; i < logs.length; i++) {
+            let habitId = logs[i].habitId;
+            let currentControl = this.habitReviewForm.get(habitId.toString());
+            currentControl?.setValue(logs[i].completed);
+          }
 
-  submit(){
+          this.newHabitLogEntry = false;
+          console.log("habit entry already exist");
+
+        } else {
+          this.newHabitLogEntry = true;
+          console.log("habit logs do not exist for current date - using default values");
+        }
+      });
+
+  }
+  checkIfGuidedJournalExist(){
+    this.accountService.getCurrentDateJournalEntry(
+      this.currentDate,
+      this.currentUserId).
+      pipe(take(1)).
+      subscribe(journalEntry => {
+        //if rating entries exist for current date
+        if (journalEntry.length !== 0) {
+          this.newJournalEntry = false;
+          this.journalRecordId = journalEntry[0].id;
+          console.log("journal entry already exist", journalEntry[0]);
+          this.guidedJournalForm.patchValue(journalEntry[0]);
+      
+        } else {
+          this.newJournalEntry = true;
+          console.log("currentdate entry does not exist");
+        }
+      });
+  }
+  submit() {
     console.log("review submit called");
+    this.submitWellnessRatingForm();
+    this.submitHabitReviewForm();
+    this.submitGuidedJournalForm();
+
+  }
+  submitGuidedJournalForm(){
+    let guidedJournalFormData: IGuidedJournalEntry = this.guidedJournalForm.getRawValue(); 
+    
+    if (this.newJournalEntry === true) {
+
+      this.newJournalEntry = false;
+
+      this.accountService.addJournalRecordEntry(guidedJournalFormData)
+      .pipe(take(1))
+      .subscribe();
+    } else if (this.newWellnessEntry === false) {
+      console.log("new entry false");
+
+      this.accountService.getCurrentDateWellnessEntry(
+        this.currentDate,
+        this.currentUserId
+      ).
+        pipe(take(1)).
+        subscribe(guidedJournalEntry => {
+
+          if (guidedJournalEntry.length !== 0) {
+            //update the existing currentDate wellness entry with the Forms data
+            this.journalRecordId = guidedJournalEntry[0].id;
+            guidedJournalFormData.id = this.journalRecordId;
+            this.accountService.updateGuidedjournalData(guidedJournalFormData).
+              pipe(take(1)).
+              subscribe(() => {
+                console.log("jounral form update success");
+                this.alertService.success('Journal Entry Successfully Updated');
+              });
```

Comment:
You cannot subscribe to an observable inside another subscribe. You should be using `switchMap` to do this. Write a switchMap to hold your conditional logic as well as returning either an empty observable (or something more valuable) in the case of the false condition. If it's true and you need to do this other observable, return the observable from the switchMap.